packages/cli/src/fulltext/AppMapIndex.ts

```diff
@@ -72,8 +76,12 @@ async function buildDocument(metadataFile: string): Promise<any> {
     });
   }
 
+  let appmapId = indexDir;
+  if (appmapId.startsWith(directory)) appmapId = appmapId.substring(directory.length + 1);
+
+  const id = packRef(directory, appmapId);
```

Comment:
Store both the directory and the full appmap path in the `id`. This seems to be the best/only way to have non-indexed metadata fields.